lib/logstash/inputs/google_pubsub.rb

```diff
@@ -228,6 +228,7 @@ def run(queue)
     end # if !@subscription
 
     @logger.debug("Pulling messages from sub '#{subscription}'")
+    codec_instance = @codec.clone
```

Comment:
I don't see a need here to clone the codec, we can use `@codec` directly in line 257